jwql/database/database_interface.py

```diff
@@ -388,6 +389,22 @@ class : obj
     return type(class_name, (base,), data_dict)
 
 
+def set_read_permissions():
+    """Set read permissions for db tables
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    None
+    """
+    with engine.connect() as conn:
+        raw_sql = text("""GRANT SELECT ON ALL TABLES IN SCHEMA public TO jwqldb_user_read;""")
```

Comment:
I would suggest to read in the database username from the `config.json` file, as to not expose usernames in our code base.  The problem though is that the `database`.`user` key may point to either the read or the write account, so one would have to do some finagling.  Perhaps something like this would work:

```python
db_username = get_config()['database']['user']
db_username = '_'.join(db_username.split('_')[:-1])
db_account = '{}_read'.format(db_username)
raw_sql = text(GRANT SELECT ON ALL TABLES IN SCHEMA public TO {};""".format(db_account)
```

Reply:
@bourque I tried this and it works, but then breaks. You never want to use `.format()` when inserting text into a sql query. So I went ahead and tried to do it "the right way"


```
def set_read_permissions(username):
    """Set read permissions for db tables"""

    with engine.connect() as conn:
        raw_sql = text("GRANT SELECT ON ALL TABLES IN SCHEMA public TO :user")
        conn.execute(raw_sql, user=username)
```

But `username` is passed as a string and the database isn't a very big fan of:

`GRANT SELECT ON ALL TABLES IN SCHEMA public TO 'jwqldb_user_read'`

and is fine with:

`GRANT SELECT ON ALL TABLES IN SCHEMA public TO jwqldb_user_read`

That extra set of `' '` seems to be whats breaking it.

Reply:
Ahh, I see.  I think you had explained this to me earlier too, sorry I misunderstood.  I'll try to think if a way around this.

Reply:
Yeah, I find it strange that the command you sent me from the ticket also contained the `' '` around the username. I am unsure if it is a `postgres` or `sqlalchemy` thing.